adaptations/Roland_JD800.py

```
@@ -0,0 +1,312 @@
+def blankedOut(message):
+    # Program dumps also match isEditBufferDump() under consecutive addressing,
+    # so fingerprint them with the program-dump blankout table first.
+    if jd_800.isSingleProgramDump(message):
+        return jd_800._apply_blankout(message.copy(), jd_800.program_dump.blank_out_zones)
+    if jd_800.isEditBufferDump(message):
+        return jd_800._apply_blankout(message.copy(), jd_800.edit_buffer.blank_out_zones)
+    raise Exception("Only works with edit buffers and program dumps")
+
+
+def calculateFingerprint(message):
+    return hashlib.md5(bytearray(blankedOut(message))).hexdigest()
```

Ruff (0.15.11)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)